### PR TITLE
Reduced Size of Pictures + Refactored bluetooth.h

### PR DESCRIPTION
**Pull request by adgsenpai**
Can your review the PR?

@jdgleaver @twinaphex

### CODE REVIEW

File: bluetooth/bluetooth_driver.c [hunk 1]
```diff
@@ -16,46 +16,59 @@
 
 #include <stdint.h>
 
-#ifdef HAVE_CONFIG_H
-#include "../config.h"
+#ifdef HAVE_CONFIG_H#include "../config.h"
+
```
Comment (RobLoach) [inline bluetooth/bluetooth_driver.c:20]: Newline change.

File: bluetooth/bluetooth_driver.c [hunk 2]
```diff
@@ -16,46 +16,59 @@
...
 #endif
 
 #include "../driver.h"
+
 #include "../list_special.h"
+
 #include "../retroarch.h"
+
 #include "../runloop.h"
+
```
Comment (RobLoach) [inline bluetooth/bluetooth_driver.c:30]: Unneeded newlines?

File: bluetooth/bluetooth_driver.c [hunk 3]
```diff
@@ -16,46 +16,59 @@
...
+  /* get_devices */
+  NULL,
+  /* device_is_connected */
+  NULL,
+  /* device_get_sublabel */
+  NULL,
+  /* connect_device */
+  NULL,
+  /* remove_device */
+  "null",
```
Comment (RobLoach) [inline bluetooth/bluetooth_driver.c:52]: These are unnecessary and confusing newlines.

File: bluetooth/bluetooth_driver.c [hunk 4]
```diff
@@ -16,46 +16,59 @@
...
-   &bluetooth_bluez,
-#endif
-#endif
-   &bluetooth_null,
-   NULL,
+const bluetooth_driver_t * bluetooth_drivers[] = {
+  #ifdef HAVE_BLUETOOTH &
+  bluetooth_bluetoothctl,
+  #ifdef HAVE_DBUS &
+  bluetooth_bluez,
```
Comment (RobLoach) [inline bluetooth/bluetooth_driver.c:59]: Having the `&` on the same line as the macro is confusing. Goes against the majority of the other RetroArch code standards.

File: bluetooth/bluetooth_driver.c [hunk 5]
```diff
@@ -67,140 +80,124 @@ bluetooth_driver_state_t *bluetooth_state_get_ptr(void)
  * Returns: string listing of all bluetooth driver names,
  * separated by '|'.
  **/
-const char* config_get_bluetooth_driver_options(void)
-{
-   return char_list_new_special(STRING_LIST_BLUETOOTH_DRIVERS, NULL);
+const char * config_get_bluetooth_driver_options(void) {
```
Comment (RobLoach) [inline bluetooth/bluetooth_driver.c:83]: RetroArch code standards use `const char*` as the type.

File: media/ico_src/icon.svg [hunk 6]
```diff
@@ -1 +1 @@
...
...2,0.7661 0.3494,1.39552 0.34977,1.39869c0.00028,0.00317 -1.17463,0.00847 -2.61111,0.01175l-2.61178,0.00597l1.83613,2.09819l1.83614,2.09819l-3.18579,-0.0028l-1.70441,-2.10229l-1.70442,-2.10229h-8.04013l-1.70442,2.10229l-1.70441,2.10229l-3.18579,0.0028zM19.97994,21.89591v-1.75143h-3.52637l-0.00608,1.72792c-0.00334,0.95036 -0.00112,1.74122 0.00519,1.75746c0.00894,0.02341 0.37533,0.02829 1.76927,0.0235l1.75799,-0.00603zM31.54644,20.14448l-3.51462,-0.01206v3.52697l3.51462,-0.01206z"/><path id="base 1 1 1" fill="none" stroke="none" stroke-miterlimit="10" stroke-width="1" d="M2,24c0,-12.15026 9.84974,-22 22,-22c12.15026,0 22,9.84974 22,22c0,12.15026 -9.84974,22 -22,22c-12.15026,0 -22,-9.84974 -22,-22z"/><rect width="44" height="22" x="2" y="2" fill="none" stroke="none" stroke-miterlimit="10" stroke-width="1"/><path fill="none" stroke="none" stroke-miterlimit="10" stroke-width="1" d="M2,24v-22h44v22z"/><path fill="none" stroke="none" stroke-miterlimit="10" stroke-width="1" d="M2,24v-22h44v22z"/><path fill="none" stroke="none" stroke-miterlimit="10" stroke-width="1" d="M2,24v-22h44v22z"/><path fill="none" stroke="none" stroke-miterlimit="10" stroke-width="1" d="M2,24v-22h44v22z"/></g></svg>
```
Comment (RobLoach) [inline media/ico_src/icon.svg:1]: Regarding the image optimization, there have been some issues in display ImgBot compressed images in RetroArch. Have some of these been tested?

File: media/icons/icon.svg [hunk 7]
```diff
@@ -1 +1 @@
...
...2,0.7661 0.3494,1.39552 0.34977,1.39869c0.00028,0.00317 -1.17463,0.00847 -2.61111,0.01175l-2.61178,0.00597l1.83613,2.09819l1.83614,2.09819l-3.18579,-0.0028l-1.70441,-2.10229l-1.70442,-2.10229h-8.04013l-1.70442,2.10229l-1.70441,2.10229l-3.18579,0.0028zM19.97994,21.89591v-1.75143h-3.52637l-0.00608,1.72792c-0.00334,0.95036 -0.00112,1.74122 0.00519,1.75746c0.00894,0.02341 0.37533,0.02829 1.76927,0.0235l1.75799,-0.00603zM31.54644,20.14448l-3.51462,-0.01206v3.52697l3.51462,-0.01206z"/><path id="base 1 1 1" fill="none" stroke="none" stroke-miterlimit="10" stroke-width="1" d="M2,24c0,-12.15026 9.84974,-22 22,-22c12.15026,0 22,9.84974 22,22c0,12.15026 -9.84974,22 -22,22c-12.15026,0 -22,-9.84974 -22,-22z"/><rect width="44" height="22" x="2" y="2" fill="none" stroke="none" stroke-miterlimit="10" stroke-width="1"/><path fill="none" stroke="none" stroke-miterlimit="10" stroke-width="1" d="M2,24v-22h44v22z"/><path fill="none" stroke="none" stroke-miterlimit="10" stroke-width="1" d="M2,24v-22h44v22z"/><path fill="none" stroke="none" stroke-miterlimit="10" stroke-width="1" d="M2,24v-22h44v22z"/><path fill="none" stroke="none" stroke-miterlimit="10" stroke-width="1" d="M2,24v-22h44v22z"/></g></svg>
```
Comment (RobLoach) [inline media/icons/icon.svg:1]: In addition, you should likely split these changes up into individual Pull Requests. Changing both the Bluetooth driver and optimizing the images is making a bunch of changes that can be difficult to test in two different change-sets. I'd recommend splitting the changes up into two different branches.

**Review by RobLoach — CHANGES_REQUESTED** [review]
I'd recommend splitting these two changes up into separate pull requests.

File: media/ico_src/icon.svg (hunk 6 above)
Comment (adgsenpai) [reply media/ico_src/icon.svg:1]: Thanks for your code review. When i have time i will propose the changes into separate pull requests. All of these mods have been made by bots. I was working on some private mods for RetroArch on my side in the future i will pull request this on my side. Thanks again for your time reviewing the Pull Request.

**Comment by adgsenpai** [comment]
As @RobLoach mentioned i should put this in separate branches which i will do in the future. I will now close the pull request.
Thanks again @RobLoach for reviewing the PR.

**Pull request closed without merging**